lib/ClangImporter/ClangImporter.cpp

```diff
@@ -768,73 +768,46 @@ bool ClangImporter::canReadPCH(StringRef PCHFilename) {
     return false;
 
   // FIXME: The following attempts to do an initial ReadAST invocation to verify
-  // the PCH, without affecting the existing CompilerInstance.
+  // the PCH, without causing trouble for the existing CompilerInstance.
   // Look into combining creating the ASTReader along with verification + update
   // if necessary, so that we can create and use one ASTReader in the common case
   // when there is no need for update.
-
-  CompilerInstance &CI = *Impl.Instance;
-  auto clangDiags = CompilerInstance::createDiagnostics(
-                                              new clang::DiagnosticOptions());
+  clang::CompilerInstance CI(Impl.Instance->getPCHContainerOperations(),
+                             &Impl.Instance->getModuleCache());
+  auto invocation =
+      std::make_shared<clang::CompilerInvocation>(*Impl.Invocation);
+  invocation->getFrontendOpts().DisableFree = false;
+  invocation->getPreprocessorOpts().RemappedFileBuffers.clear();
```

Comment:
Perhaps add a comment explaining why clearing RemappedFileBuffers is necessary?

Reply:
Will do.  (I just left a comment above in the review, if you want a preview.)

Reply:
Done in 178d209072e1c140d211b31e3c9edc22a835c2de.